test/Microsoft.AspNet.Mvc.Core.Test/PhysicalFileProviderResultTest.cs

```diff
@@ -0,0 +1,207 @@
+            // Act
+            await result.ExecuteResultAsync(context);
+
+            // Assert
+            sendFileMock.Verify();
+        }
+
+        [Fact]
+        public async Task ExecuteResultAsync_SetsSuppliedContentTypeAndEncoding()
+        {
+            // Arrange
+            var expectedContentType = "text/foo; charset=us-ascii";
+            var path = Path.GetFullPath(Path.Combine(".", "TestFiles", "FilePathResultTestFile_ASCII.txt"));
+            var result = new TestPhysicalFilePathResult(path, MediaTypeHeaderValue.Parse(expectedContentType)) { IsAscii = true };
+            var httpContext = new DefaultHttpContext();
+            var memoryStream = new MemoryStream();
+            httpContext.Response.Body = memoryStream;
+            var context = new ActionContext(httpContext, new RouteData(), new ActionDescriptor());
+
+            // Act
+            await result.ExecuteResultAsync(context);
+
+            // Assert
+            var contents = Encoding.ASCII.GetString(memoryStream.ToArray());
+            Assert.Equal("FilePathResultTestFile contents ASCII encoded", contents);
+            Assert.Equal(expectedContentType, httpContext.Response.ContentType);
+        }
+
+        [Fact]
+        public async Task ExecuteResultAsync_WorksWithAbsolutePaths()
+        {
+            // Arrange
+            var path = Path.GetFullPath(Path.Combine(".", "TestFiles", "FilePathResultTestFile.txt"));
+            var result = new TestPhysicalFilePathResult(path, "text/plain");
+
+            var httpContext = new DefaultHttpContext();
+            httpContext.Response.Body = new MemoryStream();
+
+            var context = new ActionContext(httpContext, new RouteData(), new ActionDescriptor());
+
+            // Act
+            await result.ExecuteResultAsync(context);
+            httpContext.Response.Body.Position = 0;
+
+            // Assert
+            Assert.NotNull(httpContext.Response.Body);
+            var contents = await new StreamReader(httpContext.Response.Body).ReadToEndAsync();
+            Assert.Equal("FilePathResultTestFile contents", contents);
+        }
+
+        [Theory]
+        [InlineData("FilePathResultTestFile.txt")]
+        [InlineData("./FilePathResultTestFile.txt")]
+        [InlineData(".\\FilePathResultTestFile.txt")]
+        [InlineData("~/FilePathResultTestFile.txt")]
+        [InlineData("..\\TestFiles/FilePathResultTestFile.txt")]
+        [InlineData("..\\TestFiles\\FilePathResultTestFile.txt")]
+        [InlineData("..\\TestFiles/SubFolder/SubFolderTestFile.txt")]
+        [InlineData("..\\TestFiles\\SubFolder\\SubFolderTestFile.txt")]
+        [InlineData("..\\TestFiles/SubFolder\\SubFolderTestFile.txt")]
+        [InlineData("..\\TestFiles\\SubFolder/SubFolderTestFile.txt")]
+        [InlineData("~/SubFolder/SubFolderTestFile.txt")]
+        [InlineData("~/SubFolder\\SubFolderTestFile.txt")]
+        public async Task ExecuteAsync_ThrowsFileNotFound_ForNonRootedPaths(string path)
```

Comment:
Isn't this whatever exception `GetFileStream` throws? If there was a file at `./Properties/FilePathResultTestFile.txt`- would this still throw a `FileNotFound`?